Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+using Witsml.Data;
+using Witsml.Data.Curves;
+using Witsml.Extensions;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Models.Reports;
+using WitsmlExplorer.Api.Query;
+using WitsmlExplorer.Api.Services;
+using Index = Witsml.Data.Curves.Index;
+
+namespace WitsmlExplorer.Api.Workers;
+
+/// <summary>
+/// Worker for analyzing gaps for curves of log data.
+/// </summary>
+public class AnalyzeGapWorker : BaseWorker<AnalyzeGapJob>, IWorker
+{
+    private const string DataSeparator = ",";
+    public JobType JobType => JobType.AnalyzeGaps;
+    public AnalyzeGapWorker(ILogger<AnalyzeGapJob> logger, IWitsmlClientProvider witsmlClientProvider) : base(witsmlClientProvider, logger) { }
+    
+    /// <summary>
+    /// Find all gaps for selected mnemonics and required size of gap. If isn't select mnemonics, then are used all mnemonics on log for finding gaps. 
+    /// </summary>
+    /// <param name="job">Job model of logObject, array of mnemonics, gapSize...</param>
+    /// <returns>Task of workerResult with gap report items.</returns>
+    public override async Task<(WorkerResult, RefreshAction)> Execute(AnalyzeGapJob job)
+    {
+        Logger.LogInformation("Analyzing gaps started. {jobDescription}", job.Description());
+
+        string indexCurve = job.LogReference.IndexCurve;
+        string logUid = job.LogReference.Uid;
+        bool isDepthLog = job.LogReference.IndexType == WitsmlLog.WITSML_INDEX_TYPE_MD;
+        List<AnalyzeGapReportItem> gapReportItems = new();
+        
+        WitsmlLogs logQuery = LogQueries.GetLogContent(job.LogReference.WellUid, job.LogReference.WellboreUid, logUid, job.LogReference.IndexType, job.Mnemonics, null, null);
```

Comment:
Witsml servers have a limit of how many rows can be returned at the same time, called maxDataNodes.
Using LogQueries.GetLogContent will fetch only the first maxDataNodes amount of rows.
LogDataReader can be used to read all rows of a log.

Reply:
fixed